kernel/src/process/clone.rs

```diff
@@ -202,19 +202,36 @@ fn clone_child_task(
     parent_context: &UserContext,
     clone_args: CloneArgs,
 ) -> Result<Arc<Task>> {
+    let clone_flags = clone_args.flags;
+
+    // This combination is not valid, according to the Linux man pages. See
+    // <https://www.man7.org/linux/man-pages/man2/clone.2.html>.
+    if !clone_flags.contains(CloneFlags::CLONE_VM | CloneFlags::CLONE_SIGHAND) {
+        return_errno_with_message!(
+            Errno::EINVAL,
+            "`CLONE_THREAD` without `CLONE_VM` and `CLONE_SIGHAND` is not valid"
+        );
+    }
+
+    // This is valid combination in Linux. But we do not support it yet.
+    if !clone_flags.contains(CloneFlags::CLONE_FILES | CloneFlags::CLONE_FS) {
```

Comment:
Not sure if I understand your comments correctly, but it seems that the code you provided has the same semantics as the original code?

```rust
clone_flags.contains(CloneFlags::CLONE_FILES | CloneFlags::CLONE_FS)
```
<=>
```rust
clone_flags.contains(CloneFlags::CLONE_FILES) && clone_flags.contains(CloneFlags::CLONE_FS)
```

So,
```rust
!clone_flags.contains(CloneFlags::CLONE_FILES | CloneFlags::CLONE_FS)
```
<=>
```rust
!clone_flags.contains(CloneFlags::CLONE_FILES) || !clone_flags.contains(CloneFlags::CLONE_FS)
```

Reply:
Oh, my mistake. The original version conveys the same meaning as mine.

But I actually want to point out that the flags `CLONE_VM` and `CLONE_SIGHAND` have a dependency relationship (according to the man page, "Since Linux 2.6.0, the flags mask must also include CLONE_VM if CLONE_SIGHAND is specified.") that `CLONE_FILES` and `CLONE_FS` do not have.

While the code in L209 and L217 looks the same, it's a little bit confused. So maybe the formation `!clone_flags.contains(CloneFlags::CLONE_FILES) || !clone_flags.contains(CloneFlags::CLONE_FS)` is less confused? If not, at least change the word "**and**" to "**or**" in the error message to show they're independent flags.